File: src/orchestration/DotNet/OrchestratorResult.cs
using Services.DotNet;

namespace Orchestration.DotNet;

public sealed record RetrievedRecipeInfo(string RecipeId, string Title, double SimilarityScore);

public sealed record OrchestratorResult(
    string FinalResponse,
    string SelectedAgent,
    IReadOnlyList<string> ExecutedTools,
    IReadOnlyList<RetrievedRecipeInfo> RetrievedRecipes,
    IReadOnlyList<string> ExecutionSteps,
    TimeSpan ExecutionDuration,
    IReadOnlyList<string> Errors,
    IReadOnlyList<ConversationMessage> MessagesToPersist);

File: src/orchestration/Python/orchestrator_result.py
from __future__ import annotations

from dataclasses import dataclass
from typing import Any


@dataclass(frozen=True)
class RetrievedRecipeInfo:
    recipeId: str
    title: str
    similarityScore: float


@dataclass(frozen=True)
class OrchestratorResult:
    final_response: str
    selected_agent: str
    executed_tools: list[str]
    retrieved_recipes: list[RetrievedRecipeInfo]
    execution_steps: list[str]
    execution_duration: float
    errors: list[str]
    messages_to_persist: list[Any]